docs/contributing-to-airbyte/writing-docs.md

```diff
@@ -84,6 +84,72 @@ To make complex changes or edit multiple files, edit the files on your local mac
 
 5. Assign `airbytehq/docs` as a Reviewer for your pull request. 
 
+### Custom markdown extensions for connector docs
+Airbyte's markdown documentation—particularly connector-specific documentation—needs to
+gracefully support multiple different contexts: key details may differ between open-source builds
+and Airbyte Cloud, and the more exhaustive explanations appropriate for
+https://docs.airbyte.com may bury key details when rendered as inline documentation within
+the Airbyte application. In order to support all these different contexts
+without resorting to multiple overlapping files that must be maintained in parallel,
+Airbyte's documentation tooling supports multiple nonstandard features. Please familiarize
+yourself with all the tools available to you when writing documentation for a connector,
+so that you can provide appropriately tailored information to your readers in whichever
+context they see it. As a general rule, features that introduce new behavior or prevent
+certain content from rendering will affect how the Airbyte UI displays markdown content,
+but have no impact on https://docs.airbyte.com.
+
+#### Jump to the relevant documentation section when specific Connector Builder inputs are focused with `<FieldAnchor>`
+In the documentation, the relevant section needs to be wrapped in a `<FieldAnchor field="path.to.field" />` component. These are rendered as regular divs in the documentation site, so they have no effect in places other than the in-app documentation panel—however, note that there must be blank lines between a custom tag like `FieldAnchor` the content it wraps for the documentation site to render markdown syntax inside the custom tag to html.
```

Comment:
Could we add a note here indicating that it is the field path underneath connectionSpecification, to make this very clear? Otherwise reader's may be unsure if they should include that in the path